data-reconciliation-app/docs/adr/01-data-ingest.md

```diff
@@ -0,0 +1,139 @@
+# Data Ingest API
+
+## Status
+
+Proposed
+
+## Context
+
+We need to build an API that ingests member data into our system. Once ingested, we also need to store this data in the K-V store.
+
+Gathered requirements:
+
+- Members will upload data via csv
+- Members will upload multiple records (batch) or single records via csv
+- Data will have one attribute, and we will only work with string attributes for now.
+
+## Decision
+
+### Endpoint
+
+```
+POST  /csv
+// Endpoint ingests data into data reconciliation app
+	• URL Params
+        None
+	• Headers
+        Content-Type: application/json
```

Comment:
I think this is a file upload operation and `Content-Type` will be  `multipart/form-data`.

Also, for the security of this endpoint, are we going to request a token or an API key/secret in the headers?
Maybe an API key is simpler at this stage?

```
x-api-key: {UUID}
```

Reply:
I think we need to consider one the available authenication options provided by CCF https://microsoft.github.io/CCF/main/build_apps/auth/index.html

Reply:
added to latest commit